### PR TITLE
Code quality: Simplify handling of dropped files

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
Items resolved / related issues by this PR.

Removes (looks like) unnecessary RemoteDataObject.cs & CheckDragNeedsFulltrust() method.
To be considered after PR #10925 is merged.

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

File: src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs [hunk 1]
```diff
@@ -737,35 +710,13 @@ private static async Task<(List<FileNameConflictResolveOptionType> collisions, b
...
-					return true;
-				}
-				catch (Exception ex)
-				{
-					App.Logger.Warn(ex, ex.Message);
-					return false;
-				}
-			}
-			return false;
+			return packageView is not null && (packageView.Contains(StandardDataFormats.StorageItems) || packageView.Contains("FileDrop"));
```
Comment (jiejasonliu) [inline src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs:713]: accidentally merged `packageView.Contains("FileDrop")` back in?

Comment (gave92) [reply src/Files.App/Filesystem/FilesystemOperations/Helpers/FilesystemHelpers.cs:713]: On purpose, this is the real FileDrop.

**Review by jiejasonliu — APPROVED** [review]
lgtm!

**Review by lukeblevins — APPROVED** [review]
👍🏻

**Comment by gave92** [comment]
Yeah if approvals from @jiejasonliu and luke are valid :)
Edit: if possible lets not skip insider testing for this king of changes!

**Pull request merged**